### DISCUSSION TITLE
Dropdown for Selecting Base URL which eventually will be used for the test requests

### DISCUSSION THREAD

**Discussion by FotieMConstant** [scalar/scalar#366] [Ideas]
### Description:
**Problem:**
Currently, scalar lacks a user-friendly way to select a Base URL when viewing the documentation. This can be inconvenient, especially for developers who have multiple base URLs in their documentation (e.g., dev server and production server).

**Current feature:**
What is currently done is that, when we click on a base URL it copies it to the clipboard. Which is great, but what if we want to test our local api version directly from scalar?

**Suggested Feature:**
It would be beneficial to implement a feature that allows users to select the Base URL from a dropdown menu. This dropdown menu should contain all available Base URLs within the Swagger spec. When a user selects a specific Base URL, it should automatically set the chosen URL in the entire doc, making it easier to switch between different base urls (e.g., dev server and prod server).

**Use Case:**
This feature would be particularly useful for developers who frequently work in different environments. For example, they might need to switch between a development server and a production server while testing their APIs. Having a dropdown to select the Base URL would streamline this process and improve the user experience.

For additional context, here is a screenshoot from swagger.

<img width="1269" alt="image" src="https://github.com/scalar/scalar/assets/42372656/e1fbebce-96f4-44aa-9c73-c97a2322b7aa">

---

Changing the base url from the dropdown menu equally changes the base url from the sample code and test code as well:
<img width="1265" alt="image" src="https://github.com/scalar/scalar/assets/42372656/2d1ccb2f-d122-44f2-bea2-0f1cc250e79e">

**Comment by hanspagel** [comment]
Thanks for the feedback! And what a great timing. We’re already working on it: https://github.com/scalar/scalar/pull/348 😇

**Comment by marclave** [comment]
awesome work! this was fixed in #348
